### PR TITLE
8265435: Remove dummy lists in G1CalculatePointersClosure

### PR DESCRIPTION
**Pull request by albertnetymk**
Remove the dummy list and some general cleanup.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8265435](https://bugs.openjdk.java.net/browse/JDK-8265435): Remove dummy lists in G1CalculatePointersClosure

### Reviewers
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**) ⚠️ Review applies to 6f3b9f83626d86dbf7ad7e4fda2763f111871f9e
 * [Thomas Schatzl](https://openjdk.java.net/census#tschatzl) (@tschatzl - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3568/head:pull/3568` \
`$ git checkout pull/3568`

Update a local copy of the PR: \
`$ git checkout pull/3568` \
`$ git pull https://git.openjdk.java.net/jdk pull/3568/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3568`

View PR using the GUI difftool: \
`$ git pr show -t 3568`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3568.diff">https://git.openjdk.java.net/jdk/pull/3568.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/gc/g1/g1FullGCPrepareTask.cpp [hunk 1]
```diff
@@ -39,6 +39,17 @@
 #include "oops/oop.inline.hpp"
 #include "utilities/ticks.hpp"
 
+template<bool is_humongous>
+void G1FullGCPrepareTask::G1CalculatePointersClosure::free_region(HeapRegion* hr) {
```
Comment (kstefanj) [inline src/hotspot/share/gc/g1/g1FullGCPrepareTask.cpp:43]: Since this is only for "pinned regions", maybe call this `free_pinned_region()`.

**Review by kstefanj — APPROVED** [review]
Nice cleanup. Just a small naming suggestion :)

File: src/hotspot/share/gc/g1/g1FullGCPrepareTask.cpp (hunk 1 above)
Comment (albertnetymk) [reply src/hotspot/share/gc/g1/g1FullGCPrepareTask.cpp]: Makes sense; thanks for the suggestion. Fixed.

**Review by tschatzl — APPROVED** [review]
Lgtm.

**Comment by albertnetymk** [comment]
Thanks for the review.

/integrate

**Pull request closed without merging**